trio/_core/_multierror.py

```diff
@@ -210,41 +230,72 @@ def __new__(cls, exceptions, *, _collapse=True):
             # If this lone object happens to itself be a MultiError, then
             # Python will implicitly call our __init__ on it again.  See
             # special handling in __init__.
-            return exceptions[0]
+            single = exceptions[0]
+            assert isinstance(single, cls)
```

Comment:
No, this assert is wrong. If `_collapse=True`, `MultiError([ValueError()])` just returns the `ValueError` directly, unwrapping it. Hence why I suggested the overload for `__new__()`.

Reply:
Changed in [e0f7a9c](https://github.com/python-trio/trio/pull/2742/commits/e0f7a9cac32647c303af28f101600ade0331a45e)